tests/integration/components/spread-test.js

```diff
@@ -1,30 +1,235 @@
-/* jshint expr:true */
-import { expect } from 'chai'
-import {
-  describeComponent,
-  it
-} from 'ember-mocha'
+import Ember from 'ember'
+import {expect} from 'chai'
+import {$hook, initialize as initializeHook} from 'ember-hook'
+import {describeComponent, it} from 'ember-mocha'
 import hbs from 'htmlbars-inline-precompile'
+import {beforeEach, describe} from 'mocha'
+import sinon from 'sinon'
+import SpreadMixin from 'ember-spread'
 
-describeComponent(
-  'spread',
-  'Integration: SpreadComponent',
-  {
-    integration: true
-  },
-  function () {
-    it('renders', function () {
-      // Set any properties with this.set('myProperty', 'value');
-      // Handle any actions with this.on('myAction', function(val) { ... });
-      // Template block usage:
-      // this.render(hbs`
-      //   {{#spread}}
-      //     template content
-      //   {{/spread}}
-      // `);
-
-      this.render(hbs`{{spread}}`)
-      expect(this.$()).to.have.length(1)
-    })
-  }
-)
+import {integration} from 'dummy/tests/helpers/ember-test-utils/describe-component'
+
+describeComponent(...integration('spread'), function () {
+  let handler
+
+  beforeEach(function () {
+    initializeHook()
+
+    this.registry.register('component:spread-test', Ember.Component.extend(SpreadMixin, {
+
+      // == Properties ============================================================
+
+      hook: 'spread-test',
+      layout: hbs`
+        <div data-test={{hook 'spread-property'}}>
```

Comment:
We use `-` to delimit sub-properties within our components a lot:
```hbs
{{my-foo
  hook=(concat hook '-foo')
}}
```

Because of that, I think it would be a good idea for us to avoid using `-` in our hook names. I.e. use `spreadProperty` instead of `spread-property`.  I've brought that up within my team, but probably worth bringing it up with the wider `#frost-design` group and establishing a convention.